### PR TITLE
Fix reference for GitHub

### PR DESCRIPTION
**Pull request by zregvart**
References on GitHub start with `#`[1].

[1] https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls

### CODE REVIEW

File: commit-msg/check-commit-message [hunk 1]
```diff
@@ -12,7 +12,7 @@ fi
 # The regex pattern
 # The (?i) is for case-insensitive matching
 # The \s* at the end is for optional trailing whitespace
-pattern="^(?i)(ref|refs|reference|references|res|resolve|resolves)[ \t]*:[ \t]*(gh-|\!|[A-Za-z]+-)\d+\s*$"
+pattern='^(?i)(ref|refs|reference|references|res|resolve|resolves)\s*:\s*(#\d+||[A-Za-z]+-\d+)\s*$'
```
Comment (robnester-rh) [inline commit-msg/check-commit-message]: Let's leave the `gh-` as well, just in case.

**Review by robnester-rh — CHANGES_REQUESTED** [review]
Let's leave `gh-` as well.

**Comment by zregvart** [comment]
@robnester-rh done

**Review by robnester-rh — APPROVED** [review]
LGTM

**Pull request merged**